rtcpeerconnection.go

```diff
@@ -38,17 +37,40 @@ func (t TrackType) String() string {
 	}
 }
 
+type RTCCredentialType string
+
+var (
+	RTCCredentialTypePassword RTCCredentialType = "password"
+)
+
+type RTCICEServer struct {
+	CredentialType RTCCredentialType
+	URLs           []string
+	Username       string
+	Credential     string
+}
+
+type RTCConfiguration struct {
+	ICEServers []RTCICEServer // An array of RTCIceServer objects, each describing one server which may be used by the ICE agent; these are typically STUN and/or TURN servers. If this isn't specified, the ICE agent may choose to use its own ICE servers; otherwise, the connection attempt will be made with no STUN or TURN server available, which limits the connection to local peers.
+}
+
+// New creates a new RTCPeerConfiguration with the provided configuration
+func New(config *RTCConfiguration) (*RTCPeerConnection, error) {
+	return &RTCPeerConnection{
+		config: config,
+	}, nil
+}
+
 // RTCPeerConnection represents a WebRTC connection between itself and a remote peer
 type RTCPeerConnection struct {
 	Ontrack                    func(mediaType TrackType, buffers <-chan *rtp.Packet)
 	LocalDescription           *sdp.SessionDescription
 	OnICEConnectionStateChange func(iceConnectionState ice.ConnectionState)
 
+	config *RTCConfiguration
 	tlscfg *dtls.TLSCfg
 
-	iceUsername string
```

Comment:
This will need to stay (but I gave it an awful name, so renaming it would be great). This is the `ice-ufrag` and `ice-pwd` and there is only one per session. This isn't the username/password for a TURN server.

This is used for STUN binding requests, and all your candidates will share the same one.

Reply:
I've renamed it to `iceUfrag` and `icePwd` to be close to the RFC